modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -1070,7 +1070,7 @@ private CompletableFuture<?> createTableLocally(
             if (partitionAssignments(vaultManager, tableId, 0) != null) {
                 assignmentsFuture = completedFuture(tableAssignments(vaultManager, tableId, zoneDescriptor.partitions()));
             } else {
-                assignmentsFuture = distributionZoneManager.dataNodes(causalityToken, zoneId)
+                assignmentsFuture = distributionZoneManager.dataNodes(causalityToken, catalogVersion, zoneId)
```

Comment:
It might be a bit fragile. Let's assume that tableManager.start() is partially asynchronous. In that case we may start new events processing while recovery on restart isn't fully finished. Generally speaking it'll be a disaster. But within given context we may finish with non-matching token and catalogVersion.

Reply:
From my point of view, we must forbid such situations